### PR TITLE
add code scan function

### PR DESCRIPTION
**Pull request by WenjiaoYue**
## Type of Change
Add code scan function, including bandit, pylint, pyspelling, and copyRight
## Description

detail description 
Detection of code before merging, including bandit, pylint, pyspelling, and copyRight. If there is no issue with the code scan, the test is passed. And output the corresponding log file through acrtifact. If it fails, an error will be displayed. Click the details to find the corresponding information from the .log file.

### CODE REVIEW

File: .azure-pipelines/code-scan.yml [hunk 1]
```diff
@@ -47,164 +47,194 @@ stages:
...
               targetType: "inline"
               script: |
-                docker run --disable-content-trust --privileged --name="bandit" -v ${BUILD_SOURCESDIRECTORY}:/neural-compressor ${IMAGE_NAME}:${IMAGE_TAG} bash /neural-compressor/$(CODE_SCAN_PATH)/bandit/bandit.sh $(CURRENT_STATUS)
-                echo "------$(CURRENT_STATUS)-------"
-                echo $(CURRENT_STATUS)
+                docker run --disable-content-trust --privileged --name="bandit" -v ${BUILD_SOURCESDIRECTORY}:/neural-compressor ${IMAGE_NAME}:${IMAGE_TAG} bash /neural-compressor/$(CODE_SCAN_PATH)/bandit/bandit.sh
                 
-                echo "---$(ER)----$(QR)------"
+                if [[ $? -ne 0 ]]; then          
+                    echo "##vso[task.setvariable variable=BANDIT_FALSE_STATUS]true"
```
Comment (chensuyue) [inline .azure-pipelines/code-scan.yml:53]: What does this used for?

Comment (chensuyue) [reply .azure-pipelines/code-scan.yml:53]: ok, I know it's for stop early, does it work?

Comment (WenjiaoYue) [reply .azure-pipelines/code-scan.yml:53]: The purpose of this variable is to determine if the script outputs an error, without affecting the normal execution of subsequent tasks (.log file output to artifact), and to output an error message.

Comment (WenjiaoYue) [reply .azure-pipelines/code-scan.yml:53]: > ok, I know it's for stop early, does it work?

yeah! It works

**Pull request merged**